numpy/core/src/private/ufunc_override.c

```diff
@@ -32,8 +32,8 @@ has_non_default_array_ufunc(PyObject *obj)
                                                      "__array_ufunc__");
     }
 
-    /* Fast return for ndarray */
-    if ((PyObject *)Py_TYPE(obj) == ndarray) {
+    /* Fast return for ndarray and basic types */
+    if ((PyObject *)Py_TYPE(obj) == ndarray || _is_basic_python_type(obj)) {
```

Comment:
Seems to me that the problem lies in `_is_basic_python_type`, which looks like it should be evaluating `t in (int, string, bytes, ...)`, not `isinstance(t, (int, string, bytes))`.